lib/spack/llnl/util/lock.py

```diff
@@ -267,24 +340,65 @@ def acquire_write(self, timeout=None):
         """
         timeout = timeout or self.default_timeout
 
+        locktype = 'WRITE LOCK'
         if self._writes == 0:
-            self._debug(
-                'WRITE LOCK: {0.path}[{0._start}:{0._length}] [Acquiring]'
-                .format(self))
+            self._log_acquiring(locktype)
             # can raise LockError.
             wait_time, nattempts = self._lock(fcntl.LOCK_EX, timeout=timeout)
-            self._acquired_debug('WRITE LOCK', wait_time, nattempts)
             self._writes += 1
+            self._log_acquired(locktype, wait_time, nattempts)
```

Comment:
I think this can be moved inside `_lock()` (see above)

Reply:
These are actually dependent on an accurate `_writes` (or above `_reads`) count when verbose.